lib/internal/fs/watch/linux.js

```diff
@@ -0,0 +1,278 @@
+
+        if (this.#closed) {
+          break;
+        }
+
+        if (!this.#files.has(f)) {
+          this.#files.set(f, file);
+          this.emit('change', 'rename', this.#getPath(f));
+
+          if (file.isDirectory()) {
+            await this.#watchFolder(f);
+          } else {
+            this.#watchFile(f);
+          }
+        }
+      }
+    } catch (error) {
+      this.emit('error', error);
+    }
+  }
+
+  #watchFile(file) {
+    const { watchFile } = lazyLoadFsSync();
+
+    if (this.#closed) {
+      return;
+    }
+
+    const existingStat = this.#files.get(file);
+
+    watchFile(file, {
+      persistent: this.#options.persistent,
+    }, (statWatcher, previousStatWatcher) => {
+      if (existingStat && !existingStat.isDirectory() &&
+        statWatcher.nlink !== 0 && existingStat.mtime?.getTime() === statWatcher.mtime?.getTime()) {
+        return;
+      }
+
+      this.#files.set(file, statWatcher);
+
+      if (statWatcher.birthtimeMs === 0 && previousStatWatcher.birthtimeMs !== 0) {
+        // The file is now deleted
+        this.#files.delete(file);
+        this.emit('change', 'rename', this.#getPath(file));
+
+        if (statWatcher.isDirectory()) {
+          this.#unwatchFolder(file);
+        }
+      } else {
+        this.emit('change', 'change', this.#getPath(file));
+
+        if (statWatcher.isDirectory()) {
+          this.#watchFolder(file);
+        }
+      }
+    });
+  }
+
+  [kFSWatchStart](filename) {
+    if (typeof filename !== 'string' && !(filename instanceof Buffer) && !(filename instanceof URL)) {
+      throw new ERR_INVALID_ARG_TYPE('filename', 'string', filename);
+    }
+
+    if (typeof filename === 'string') {
+      if (filename.length === 0) {
+        throw new ERR_INVALID_ARG_TYPE('filename', 'string', filename);
+      }
+    } else if (Buffer.isBuffer(filename)) {
+      validateBuffer(filename, 'filename');
+
+      if (filename.length === 0) {
+        throw new ERR_INVALID_ARG_TYPE('filename', 'buffer', filename);
+      }
+    }
+
+    try {
+      const file = lazyLoadFsSync().statSync(filename);
+
+      this.#rootPath = filename;
+      this.#closed = false;
+      this.#files.set(filename, file);
+
+      if (file.isDirectory()) {
+        traverse(filename, this.#files)
+          .then(() => {
+            for (const f of this.#files.keys()) {
+              this.#watchFile(f);
+            }
+          });
+      } else {
+        this.#watchFile(filename);
+      }
+    } catch (error) {
+      if (error.code === 'ENOENT') {
+        error.filename = filename;
+        throw error;
+      }
+    }
+
+  }
+
+  ref() {
+    // This is kept to have the same API with FSWatcher
+    throw new ERR_FEATURE_UNAVAILABLE_ON_PLATFORM('ref');
+  }
+
+  unref() {
+    // This is kept to have the same API with FSWatcher
+    throw new ERR_FEATURE_UNAVAILABLE_ON_PLATFORM('unref');
+  }
+
+  [SymbolAsyncIterator]() {
+    const watcher = this;
+    return ObjectCreate({
```

Comment:
Sorry my previous suggestion was suboptimal, we don't need `Object.create` here, we can return the object directly.
```suggestion
    return {
```

Reply:
No worries. I've updated the pull request.